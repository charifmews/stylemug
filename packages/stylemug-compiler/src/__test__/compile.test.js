const compile = require('../compile');

describe('compile', () => {
  it('should compile schema', () => {
    const result = compile.compileSchema({
      default: {
        color: 'red',
        backgroundColor: 'yellow',
      },
      foo: {
        color: 'blue',
      },
      bar: {
        color: 'red',
      },
    });

    expect(result).toMatchSnapshot();
  });

  it('should compile nested selectors', () => {
    const result = compile.compileSchema({
      default: {
        color: 'red',

        '&:focus': {
          color: 'blue',
        },
      },
    });

    expect(result).toMatchSnapshot();
  });

  it('should compile nested media query', () => {
    const result = compile.compileSchema({
      default: {
        color: 'red',

        '@media only screen and (max-width: 600px)': {
          color: 'blue',
        },
      },
    });

    expect(result).toMatchSnapshot();
  });
});
