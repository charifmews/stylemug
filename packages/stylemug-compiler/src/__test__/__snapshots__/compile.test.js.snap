// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compile should compile nested media query 1`] = `
Object {
  "default": Object {
    "c73de52f1": Object {
      "children": "",
      "key": "color",
      "keyId": "caa38a5b5",
      "media": "@media only screen and (max-width: 600px)",
      "value": "blue",
    },
    "cb34cd11f": Object {
      "children": "",
      "key": "color",
      "keyId": "c3d7e6258",
      "media": undefined,
      "value": "red",
    },
  },
}
`;

exports[`compile should compile nested selectors 1`] = `
Object {
  "default": Object {
    "c2977468e": Object {
      "children": ":focus",
      "key": "color",
      "keyId": "cc8e550a",
      "media": undefined,
      "value": "blue",
    },
    "cb34cd11f": Object {
      "children": "",
      "key": "color",
      "keyId": "c3d7e6258",
      "media": undefined,
      "value": "red",
    },
  },
}
`;

exports[`compile should compile schema 1`] = `
Object {
  "bar": Object {
    "cb34cd11f": Object {
      "children": "",
      "key": "color",
      "keyId": "c3d7e6258",
      "media": undefined,
      "value": "red",
    },
  },
  "default": Object {
    "b765dfa68": Object {
      "children": "",
      "key": "backgroundColor",
      "keyId": "bb7c5a6d0",
      "media": undefined,
      "value": "yellow",
    },
    "cb34cd11f": Object {
      "children": "",
      "key": "color",
      "keyId": "c3d7e6258",
      "media": undefined,
      "value": "red",
    },
  },
  "foo": Object {
    "cf3f0b584": Object {
      "children": "",
      "key": "color",
      "keyId": "c3d7e6258",
      "media": undefined,
      "value": "blue",
    },
  },
}
`;

exports[`compile should compile selectors with number as value 1`] = `
Object {
  "f8eb48ce2": Object {
    "children": undefined,
    "key": "fontSize",
    "keyId": "f2951bd7",
    "media": undefined,
    "value": 12,
  },
}
`;
